### DISCUSSION TITLE
New Features: Mapbox

### DISCUSSION THREAD

**Discussion by sarp86** [oguzhnatly/flutter_carplay#16] [Ideas]
Thank you for this nice project!

1. Would be nice if you could also implement mapbox map to carplay. We would be very happy. https://pub.dev/packages/mapbox_gl
2. Playing radio would be nice too

Best regards from Germany :-)

**Comment by JaffaKetchup** [comment]
Instead of using MapBox, can I suggest 'flutter_map'? It supports many more sources and much more configuration.

**Comment by sarp86** [comment]
Hey JaffaKetchup,

thank you for your fast response.

We implement mapbox to our app's, we decided to go forward with mapbox because of stability and feature set. 
It would be huge effort to change it to other map. I would pleased if you could implement mapbox too.

Thank you!

**Comment by oguzhnatly** [comment]
Thank you so much for your supports and opinions in the discussion, @sarp86 and @JaffaKetchup. I really appreciate it!

First and foremost, I'd like to state that this package will support as many packages as possible at the same time. That's why you shouldn't be concerned 😊. However, that I'm the only developer of this project, it may take some time to complete all of them. I'm currently working on the CarPlay Voice Control and Now Playing Music System. So that, you will be able to use voice control, text-to-speech, speech-to-text, and **play music or radio**. It's nearly completed at this time.

For Map, I currently do not plan to implement flutter_map in the first plan because Native MapBox has CarPlay functionalities and it is much easier and faster to implement this package for Flutter than a flutter native map. However, this does not indicate that the flutter native map will never support this package; it will, but not in the initial plan. It makes no sense whether the flutter map package supports more sources or more configuration. CarPlay severely restricts many of its functionalities, since it has constant configurations which they are unchangeable. Consequently, **MapBox Native Package** will be the first plan to implement to the flutter_carplay package.

For now, you can track the development activity of version 1.1 [here](https://github.com/oguzhnatly/flutter_carplay/projects/1). I plan to release Mapbox implementations in version 1.2 or in its beta. If you all or anyone is interested in and wants to help and contribute, join our [Discord Flutter Auto Technologies Server](https://discord.gg/Xz6WVezFfh) or send me [an email](mailto:info@oguzhanatalay.com?subject=Contribute%20flutter_carplay%20#6%20issue) or a pull request so that we can improve all of our applications as soon as possible. 🚀

If you have any further questions or requests in this queue, you can continue to write in this issue. Greetings to Germany and UK :)

**Comment by JaffaKetchup** [comment]
I can understand that you only have you to develop the package. (I also am an independent package maintainer/development). Make sure you don't get yourself too stressed!

The new features sound cool!

I can understand also why you won't implement `flutter_map`, was only suggesting because it is more customizable (and I develop a plugin for it, so I get those juicy GitHub stars 🤣).

Likewise, I say hello to Turkey :)

**Comment by oguzhnatly** [comment]
Quick update on this! The Voice Control and Now Playing features mentioned above are finally being finalized and will ship very soon.

As for Mapbox/Map support: this is still planned but requires the `com.apple.developer.carplay-maps` entitlement and `CPMapTemplate` integration, which is a significant piece of work. No ETA yet, but if anyone wants to contribute on this front, PRs are very welcome! 🗺️
